From ef8159c07fe79d8066d8a0e1116a16e381485f79 Mon Sep 17 00:00:00 2001
From: "env." <162432464+belowzeroiq@users.noreply.github.com>
Date: Fri, 25 Jul 2025 22:32:20 +0700
Subject: [PATCH] treewide: clang compilation fixup

this fix the error that present on clang 21,22 and so on.
---
 arch/arm64/kvm/sys_regs.c |  2 +-
 drivers/tty/sysrq.c       | 12 ++++++++++++
 2 files changed, 13 insertions(+), 1 deletion(-)

diff --git a/arch/arm64/kvm/sys_regs.c b/arch/arm64/kvm/sys_regs.c
index 935b76225086..bd53880c60ca 100644
--- a/arch/arm64/kvm/sys_regs.c
+++ b/arch/arm64/kvm/sys_regs.c
@@ -2790,7 +2790,7 @@ int kvm_arm_copy_sys_reg_indices(struct kvm_vcpu *vcpu, u64 __user *uindices)
 void kvm_sys_reg_table_init(void)
 {
 	unsigned int i;
-	struct sys_reg_desc clidr;
+	struct sys_reg_desc clidr = {};
 
 	/* Make sure tables are unique and in order. */
 	BUG_ON(check_sysreg_table(sys_reg_descs, ARRAY_SIZE(sys_reg_descs), false));
diff --git a/drivers/tty/sysrq.c b/drivers/tty/sysrq.c
index 47792f4ff9d3..e1da70440d61 100644
--- a/drivers/tty/sysrq.c
+++ b/drivers/tty/sysrq.c
@@ -471,7 +471,11 @@ static const struct sysrq_key_op *sysrq_key_table[62] = {
 	NULL,				/* a */
 	&sysrq_reboot_op,		/* b */
 	&sysrq_crash_op,		/* c */
+#ifdef CONFIG_LOCKDEP
 	&sysrq_showlocks_op,		/* d */
+#else
+	NULL,				/* d */
+#endif
 	&sysrq_term_op,			/* e */
 	&sysrq_moom_op,			/* f */
 	/* g: May be registered for the kernel debugger */
@@ -483,7 +487,11 @@ static const struct sysrq_key_op *sysrq_key_table[62] = {
 #else
 	NULL,				/* j */
 #endif
+#ifdef CONFIG_LOCKDEP
 	&sysrq_SAK_op,			/* k */
+#else
+	NULL,				/* k */
+#endif
 #ifdef CONFIG_SMP
 	&sysrq_showallcpus_op,		/* l */
 #else
@@ -495,7 +503,11 @@ static const struct sysrq_key_op *sysrq_key_table[62] = {
 	NULL,				/* o */
 	&sysrq_showregs_op,		/* p */
 	&sysrq_show_timers_op,		/* q */
+#ifdef CONFIG_LOCKDEP
 	&sysrq_unraw_op,		/* r */
+#else
+	NULL,				/* r */
+#endif
 	&sysrq_sync_op,			/* s */
 	&sysrq_showstate_op,		/* t */
 	&sysrq_mountro_op,		/* u */
-- 
2.50.1

